shared/pkg/families/manager.go

```diff
@@ -103,17 +114,31 @@ func (m *Manager) Run(ctx context.Context) (*results.Results, RunErrors) {
 				<-result
 				close(result)
 			}()
-			familyErrors[family.GetType()] = fmt.Errorf("failed to run family %v: aborted", family.GetType())
+			oneOrMoreFamilyFailed = true
+			if err := notifier.FamilyFinished(ctx, FamilyResult{
+				Result:     nil,
+				FamilyType: family.GetType(),
+				Err:        fmt.Errorf("failed to run family %v: aborted", family.GetType()),
+			}); err != nil {
+				log.Errorf("Failed to export family result: %v", err)
+			}
 		case r := <-result:
 			log.Debugf("received result from family %q: %v", family, r)
-			if r.err != nil {
-				familyErrors[family.GetType()] = fmt.Errorf("failed to run family %v: %w", family.GetType(), r.err)
+			if r.Err != nil {
+				oneOrMoreFamilyFailed = true
 			} else {
-				familyResults.SetResults(r.result)
+				familyResults.SetResults(r.Result)
+			}
+			if err := notifier.FamilyFinished(ctx, r); err != nil {
+				log.Errorf("Failed to export family result: %v", err)
```

Comment:
We need to make sure that if we fail to notify in progress or finished, that it counts as a overall family run failure (passed to MarkDone in the CLI), so perhaps we do need a list of errors here to return e.g.:

```
errors := []error{}
for _, family := range m.families {
	if err := notifier.FamilyStarted(ctx, family.GetType()); err != nil {
	    errors = append(errors, fmt.Errorf("family started notification failed: %v", err))
	    continue
	}
	
	// run family
	
	if err := notifier.FamilyFinished(ctx, r); err != nil {
	   errors = append(errors, fmt.Errorf("family finished notification failed: %v", err))
	}
}
return errors
```

Reply:
Thing is, if we fail to export result, we will probably also fail to MarkDone..

Reply:
we also might not fail though (vmclarity network glitch or something), and we don't want a ScanResult to be marked as "Done" with no errors when a family hasn't been left InProgress because it failed to export.